client/src/components/AddPost/AddPost.tsx

```diff
@@ -0,0 +1,195 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Modal from '@mui/material/Modal';
+import TextareaAutosize from '@mui/material/TextareaAutosize';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import ImageIcon from '@mui/icons-material/Image';
+import postSchema from '../../Validation/addPost';
+import Username from '../UserInfo';
+import BasicSelect from './BasicSelect';
+import ITag from '../../Interfaces/post/ITag';
+import IAddPost from '../../Interfaces/post/IAddPost';
+import ApiServices from '../../services/ApiService';
+import UploadPic from '../UploadPic';
+import handleUpload from '../handleUpload';
+
+const AnimalList:ITag[] = [
+  { id: 1, name: 'Cats' },
+  { id: 2, name: 'Dogs' },
+  { id: 3, name: 'Turtles' },
+
+];
+
+const TagList:ITag[] = [
+  { id: 1, name: 'Need help' },
+  { id: 2, name: 'Advise' },
+  { id: 3, name: 'Up for adoption' },
+  { id: 4, name: 'Memes' },
+  { id: 5, name: 'Up for sale' },
+];
+const style = {
+  position: 'absolute' as 'absolute',
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  justifyContent: 'space-around',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  minWidth: '55%',
+  minHeight: '55%',
+  bgcolor: 'background.paper',
+  borderRadius: '15px',
+  boxShadow: 24,
+  p: 4,
+};
+
+const AddPost = () => {
+  const [file, setFile] = React.useState<any>('');
+
+  const [percent, setPercent] = React.useState(0);
+
+  const [open, setOpen] = React.useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  const [postData, setPostData] = React.useState<IAddPost>({
+    UserId: 1,
+    content: '',
+    image: '',
+    TagId: 0,
+    AnimalId: 0,
+  });
+
+  const handleClick = async () => {
+    ApiServices.init();
+    try {
+      const result = await ApiServices.post('/posts', postData);
+      console.log(result);
+    } catch (err) {
+      console.log(err);
+    }
+  };
+  React.useEffect(() => {
+    if (postData.image) {
+      handleClick();
+      console.log(postData, '1111111111111111111111');
```

Comment:
Console logs

Reply:
the pr is still under construction : ' ) ill set it to draft